engine/src/main/java/org/hibernate/search/util/logging/impl/Log.java

```diff
@@ -736,4 +737,6 @@
 	@Message(id = 248, value = "WorkList should never be empty. Stacktrace below \n %s" )
 	void workListShouldNeverBeEmpty(String stackTrace);
 
+	@Message(id = 249, value = "Unsupported value type for configuration property " + Environment.ERROR_HANDLER + ": %1$s")
+	SearchException unsupportedErrorHandlerConfigurationValueType(@FormatWith(ClassObjectFormatter.class) Class<?> errorHandlerValueType);
```

Comment:
I like it because it gives a consistent "rendering" of types used in logs,
without the need for the calling code to do so.

For Class<?> objects the proposed impl returns the FQN (i.e. not "class
FQN" as Class#toString() does). Of course you only "gain" something when
using it across the board which is what we do in OGM. So we don't have to
remember at all call sites to consistently pass Class#getName() to a String
param but just use the class object and be done with it.